src/context.rs

```diff
@@ -532,11 +478,495 @@ pub fn get_version_str() -> &'static str {
     &DC_VERSION_STR
 }
 
+/// Represents a file in the blob directory.
+///
+/// The object has a name, which will always be valid UTF-8.  Having a
+/// blob object does not imply the respective file exists, however
+/// when using one of the `create*()` methods a unique file is
+/// created.
+#[derive(Debug)]
+pub struct BlobObject<'c> {
```

Comment:
this probably should live in its own file

Reply:
yeah, considered that too.  moved it now.